upi/aws/cloudformation/02_cluster_infra.yaml

```diff
@@ -4,9 +4,18 @@ Description: Template for Openshift Cluster UPI Network Elements (Route53 & LBs)
 Parameters:
   ClusterName:
     AllowedPattern: ^([a-zA-Z][a-zA-Z0-9\-]{0,31})$
+    MaxLength: 32
```

Comment:
One of the reasons we generate the infra name is that it allows us to abbreviate long cluster names.  So ideally we'd lift this restriction and move the `ClusterName` limits closer to the limits we apply to IPI cluster names.  Doesn't have to happen in this PR though ;).

Reply:
Yeah, I don't know the formal limits on these things (including punctuation allowed or just hyphens like we fixed in another PR). Zapping some of these later with more accuracy to real world rules would be a good scrub.